src/main/java/org/motechproject/newebodac/helper/EncryptionHelper.java

```diff
@@ -0,0 +1,75 @@
+package org.motechproject.newebodac.helper;
+
+import java.security.spec.KeySpec;
+import java.util.Base64;
+
+import javax.crypto.Cipher;
+import javax.crypto.SecretKey;
+import javax.crypto.SecretKeyFactory;
+import javax.crypto.spec.IvParameterSpec;
+import javax.crypto.spec.PBEKeySpec;
+import javax.crypto.spec.SecretKeySpec;
+import org.springframework.boot.context.properties.ConfigurationProperties;
+import org.springframework.context.annotation.Configuration;
+
+@Configuration
+@ConfigurationProperties(prefix = "encryption")
+public class EncryptionHelper {
+
+  /**
+   * Encrypts provided value.
+   */
+  public static String encrypt(String strToEncrypt) {
+
+    /**
+     * Gets required values to encryption - secret key, salt and iv.
+     */
+    String secret = "${encryption.key-crypto}";
+    String salt = "${encryption.salt}";
+    String iv = "${encryption.iv}";
+
+    try {
+      IvParameterSpec ivspec = new IvParameterSpec(iv.getBytes());
+      SecretKeyFactory factory = SecretKeyFactory.getInstance("PBKDF2WithHmacSHA256");
+      KeySpec spec = new PBEKeySpec(secret.toCharArray(), salt.getBytes(), 65536, 256);
+      SecretKey tmp = factory.generateSecret(spec);
+      SecretKeySpec secretKey = new SecretKeySpec(tmp.getEncoded(), "AES");
+
+      Cipher cipher = Cipher.getInstance("AES/CBC/PKCS5Padding");
+      cipher.init(Cipher.ENCRYPT_MODE, secretKey, ivspec);
+      return Base64.getEncoder().encodeToString(cipher.doFinal(strToEncrypt.getBytes("UTF-8")));
+    } catch (Exception e) {
+      System.out.println("Error while encrypting: " + e.toString());
```

Comment:
rethrow the exception here instead, you can use the IllegalStateException and add a message there "Error occurred while encrypting the data"

Reply:
I additionally had to add catching the UnsupportedEncodingException, because after changing Exception to GeneralSecurityException it underlined the getBytes() and requested catching this exception.